File: src/RNetPi.Core/Packets/PacketS2CMediaPlayState.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x37
/// Media Play State
/// Sends media play state of a source
/// Data:
///     (Unsigned Char) SourceID
///     (Unsigned Char) Playing / Paused
/// </summary>
public class PacketS2CMediaPlayState : PacketS2C
{
    public const byte ID = 0x37;

    public PacketS2CMediaPlayState(byte sourceID, bool isPlaying)
    {
        Writer.Write(sourceID);
        Writer.Write((byte)(isPlaying ? 0x01 : 0x00));
    }

    public override byte GetID() => ID;
}

File: src_js/server/packets/PacketS2CMediaPlayState.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x37
 * Media play state
 * Sends media play state of a source
 * Data:
 *      (Unsigned Char) SourceID
 *      (Unsigned Char) Playing / Paused
 */
class PacketS2CMediaPlayState extends PacketS2C {
    constructor(sourceID, isPlaying) {
        super();

        this._buffer.writeUInt8(sourceID);
        this._buffer.writeUInt8(isPlaying ? 0x01 : 0x00);
    }

    getID() {
        return 0x37;
    }
}

module.exports = PacketS2CMediaPlayState;
